enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx

```diff
@@ -0,0 +1,66 @@
+import { useState } from "react";
+
+import { withPublicComponentWrapper } from "embedding-sdk/components/private/PublicComponentWrapper";
+import {
+  ALL_MODELS,
+  COLLECTION_PAGE_SIZE,
+} from "metabase/collections/components/CollectionContent";
+import { CollectionItemsTable } from "metabase/collections/components/CollectionContent/CollectionItemsTable";
+import CollectionBreadcrumbs from "metabase/nav/containers/CollectionBreadcrumbs/CollectionBreadcrumbs";
+import type {
+  CollectionEssentials,
+  CollectionId,
+  CollectionItem,
+  CollectionItemModel,
+} from "metabase-types/api";
+
+type CollectionBrowserProps = {
+  collectionId?: CollectionId;
+  onClick?: (item: CollectionItem) => void;
+  pageSize?: number;
+  visibleCollectionTypes?: CollectionItemModel[];
+};
+
+export const CollectionBrowser = withPublicComponentWrapper(
+  ({
+    collectionId,
+    onClick,
+    pageSize = COLLECTION_PAGE_SIZE,
+    visibleCollectionTypes = ALL_MODELS,
+  }: CollectionBrowserProps) => {
+    const [currentCollectionId, setCurrentCollectionId] = useState(
+      collectionId ?? "root",
+    );
+
+    const onClickItem = (item: CollectionItem) => {
+      if (onClick) {
+        onClick(item);
+      }
+
+      if (item.model === "collection") {
+        setCurrentCollectionId(item.id);
+      }
+    };
+
+    const onClickBreadcrumbItem = (item: CollectionEssentials) => {
+      setCurrentCollectionId(item.id);
+    };
+
+    return (
+      <>
```

Comment:
Do you think we should but a `className` and `style` tag so users can customize how these components are oriented?